tools/ansible/tasks/wsu/test/e2e/wsu_test.go

```diff
@@ -288,3 +298,246 @@ func testHNSNetworksCreated(t *testing.T) {
+			"$$k=$$callerCounts.Get_Item($$clientIP) ;if ($$k -ne $$null) { $$count += $$k } ;" +
+			"$$callerCounts.Set_Item($$clientIP, $$count) ;$$ip=(Get-NetAdapter | " +
+			"Get-NetIpAddress); $$header='<html><body><H1>Windows Container Web Server</H1>' " +
+			";$$callerCountsString='' ;$$callerCounts.Keys | % { $$callerCountsString+='<p>IP {0}" +
+			" callerCount {1} ' -f $$ip[1].IPAddress,$$callerCounts.Item($$_) } " +
+			";$$footer='</body></html>' ;$$content='{0}{1}{2}' -f " +
+			"$$header,$$callerCountsString,$$footer ;Write-Output $$content ;$$buffer = " +
+			"[System.Text.Encoding]::UTF8.GetBytes($$content) ;$$response.ContentLength64 = " +
+			"$$buffer.Length ;$$response.OutputStream.Write($$buffer, 0, $$buffer.Length) ;" +
+			"$$response.Close() ;$$responseStatus = $$response.StatusCode ;" +
+			"Write-Host('< {0}' -f $$responseStatus)  } ; "}
+	winServerDeployment, err := createWindowsServerDeployment("win-server", winServerCommand)
+	require.NoError(t, err, "Could not create Windows deployment")
+	defer deleteDeployment(winServerDeployment.Name)
+	// Wait until the server is ready to be queried
+	winServerPod, err := waitUntilPodHasRunningContainer(*winServerDeployment.Spec.Selector)
+	require.NoError(t, err, "Could not retrieve pod with selector %v", *winServerDeployment.Spec.Selector)
+
+	// test Windows <-> Linux
+	// This will install curl and then curl the windows server.
+	linuxCurlerCommand := []string{"bash", "-c", "yum update; yum install curl -y; curl " + winServerPod.Status.PodIP}
+	linuxCurlerJob, err := createLinuxJob("linux-curler", linuxCurlerCommand)
+	require.NoError(t, err, "Could not create Linux job")
+	defer deleteJob(linuxCurlerJob.Name)
+	err = waitUntilJobSucceeds(linuxCurlerJob.Name)
+	assert.NoError(t, err, "Could not curl the Windows server from a linux container")
+
+	// test Windows <-> Windows on same node
+	// This will continually try to read from the Windows Server. We have to try multiple times as the network is flaky
+	// on container startup
+	winCurlerCommand := []string{"powershell.exe", "-command", "for (($i =0), ($j = 0); $i -lt 10; $i++) { " +
+		"$response = Invoke-Webrequest -UseBasicParsing -Uri " + winServerPod.Status.PodIP +
+		"; $code = $response.StatusCode; echo \"GET returned code $code\";" +
+		"If ($code -eq 200) {exit 0}; Start-Sleep -s 10;}; exit 1" + winServerPod.Status.PodIP}
+	winCurlerJob, err := createWindowsServerJob("win-curler", winCurlerCommand)
+	require.NoError(t, err, "Could not create Windows job")
+	defer deleteJob(winCurlerJob.Name)
+	err = waitUntilJobSucceeds(winCurlerJob.Name)
+	assert.NoError(t, err, "Could not curl the Windows server from a Windows container")
+
+	// TODO: test Windows <-> Windows on different node
+}
+
+// waitUntilJobSucceeds will return an error if the job fails or reaches a timeout
+func waitUntilJobSucceeds(name string) error {
+	var job *batchv1.Job
+	var err error
+	for i := 0; i < 10; i++ {
```

Comment:
You have code like this scattered around. Please make constants for you retries and use it everywhere.